examples/sw/simple_system/common/simple_system_common.c

```diff
@@ -139,3 +139,37 @@ void simple_exc_handler(void) {
 
   while(1);
 }
+
+volatile uint64_t time_elapsed;
+uint32_t time_increment;
+
+inline static void increment_timecmp(uint32_t time_base) {
+  uint64_t current_time = (uint64_t)DEV_READ(TIMER_BASE + TIMER_MTIMEH, 0);
+  current_time <<= 32;
+  current_time |= (uint64_t)DEV_READ(TIMER_BASE + TIMER_MTIME, 0);
```

Comment:
Similar to @vogelpi's comment the spec recommends when reading you read timeh, then time, then timeh again, checking if timeh has changed. If so you had a time -> timeh overflow so should try the whole sequence again until you succeed (unlike to have more than one retry unless there's a very slow set of interrupts in the middle!)

Reply:
Good point - now added as a separate function